main.tf

```diff
@@ -0,0 +1,86 @@
+terraform {
+    backend "azurerm" {
+      resource_group_name   = "AmericanExpress1_RyanSharp_ProjectExercise"
+      storage_account_name  = "rysharpstorage"
+      container_name        = "tododb"
+      key                   = "terraform.tfstate"
+    }
+    required_providers {
+        azurerm = {
+            source = "hashicorp/azurerm"
+            version = ">= 2.49"
+        }
+    }
+}
+provider "azurerm" {
+    features {}
+}
+data "azurerm_resource_group" "main" {
+    name = "AmericanExpress1_RyanSharp_ProjectExercise"
+}
+
+resource "azurerm_app_service_plan" "main" {
+    name = "terraformed-asp"
+    location = var.LOCATION
+    resource_group_name = data.azurerm_resource_group.main.name
+    kind = "Linux"
+    reserved = true
+    sku {
+        tier = "Basic"
+        size = "B1"
+    }
+}
+
+resource "azurerm_app_service" "main" {
+    name = "rysharp-terraform-todoapp"
+    location = var.LOCATION
+    resource_group_name = data.azurerm_resource_group.main.name
+    app_service_plan_id = azurerm_app_service_plan.main.id
+    site_config {
+        app_command_line = ""
+        linux_fx_version = "DOCKER|rysharp/todo-app:latest"
+    }
+    app_settings = {
+        "DATABASE_URL" = "mongodb://${azurerm_cosmosdb_account.maindbaccount.name}:${azurerm_cosmosdb_account.maindbaccount.primary_key}@${azurerm_cosmosdb_account.maindbaccount.name}.mongo.cosmos.azure.com:10255/DefaultDatabase?ssl=true&replicaSet=globaldb&retrywrites=false&maxIdleTimeMS=120000"
+        "DATABASE_NAME" = "tododb"
+        "DOCKER_REGISTRY_SERVER_URL" = "https://index.docker.io"
+        "CLIENT_ID" = var.CLIENT_ID
+        "CLIENT_SECRET" = var.CLIENT_SECRET
+        "DOCKER_ENABLE_CI" = "true"
+        "DOCKER_REGISTRY_SERVER_URL" = "https://index.docker.io/v1"
+        "FLASK_APP" = "todo_app.app"
+        "FLASK_ENV" = "production"
+        "SECRET_KEY" = "Shhh"
```

Comment:
I'm sure you're aware that this should be a variable like CLIENT_SECRET, but just checking...

I'm fine with you not bothering for this exercise, but in practice leaking this means that anyone could fake a login session cookie.